### PR TITLE
Make site message bar optional

### PR DESCRIPTION
**Pull request by jordandrako**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

1. Moves `SiteMessageBar` to example-app-base
2. Moves the `renderSiteMessageBar` to the `SiteDefinition` so it's optional. I consider a band-aid until we come up with a way to cleanly do this in `INavPage`.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8986)

### CODE REVIEW

File: packages/example-app-base/src/components/SiteMessageBar/SiteMessageBar.tsx [hunk 1]
```diff
@@ -1,33 +1,13 @@
...
-  /**
-   * Prefix for session storage key for storing if the message bar has been closed.
-   * If not provided, local storage will not be used.
-   */
-  sessionStoragePrefix?: string;
-
-  /** Styles for the message bar */
-  styles?: IStyleFunctionOrObject<IMessageBarStyleProps, IMessageBarStyles>;
-}
+import { MessageBar } from 'office-ui-fabric-react/lib/MessageBar';
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/SiteMessageBar/SiteMessageBar.tsx]: If this is going to be an independent component, I don't think it should be called SiteMessageBar. I thought about calling it just MessageBar instead, but having component names that duplicate built-in Fabric components isn't great. Maybe Banner would work?

File: apps/fabric-website/src/SiteDefinition/SiteDefinition.tsx [hunk 2]
```diff
@@ -76,3 +77,25 @@ export const SiteDefinition: ISiteDefinition<Platforms> = {
     '#/styles/utilities': '#/styles/web'
   }
 };
+
+function _renderSiteMessageBar(pagePath?: string): JSX.Element | undefined {
```
Comment (ecraig12345) [inline apps/fabric-website/src/SiteDefinition/SiteDefinition.tsx]: This seems a lot more awkward and hack-ish than having Site render the message bar. A somewhat better design might be to put the URLs that should show a message bar and the corresponding props in the site definition:
```ts
// (I don't like this interface name but can't think of a better one right now.)
interface IMessageBarConfig {
  url: string | RegExp;
  messageBarProps: ISiteMessageBarProps;
}

interface ISiteDefinition {
  /**
   * If the route path matches any of the strings or regular expressions specified in
   * an entry's `url` property, show a message bar with the corresponding props.
   */
  messageBars?: IMessageBarConfig[];
}
```

File: packages/example-app-base/src/components/SiteMessageBar/SiteMessageBar.tsx (hunk 1 above)
Comment (jordandrako) [reply packages/example-app-base/src/components/SiteMessageBar/SiteMessageBar.tsx]: I don't like the idea of having duplicate names either (our Table components, ugh). SiteMessage or SiteMessageBar make the most sense in my opinion, but maybe it should stick to the fabric-website directory so its purpose is more explicit?

File: apps/fabric-website/src/SiteDefinition/SiteDefinition.tsx (hunk 2 above)
Comment (jordandrako) [reply apps/fabric-website/src/SiteDefinition/SiteDefinition.tsx]: It is a little hacky, I did a little exploration of just defining the props in an object per page, but it was going to take some refactoring of getting the page props in `Site` in order for the `renderSiteMessageBar` method to get those props from sub pages. It worked on top level pages like controls and home page fine though.

File: apps/fabric-website/src/SiteDefinition/SiteDefinition.tsx [hunk 3]
```diff
@@ -55,5 +55,22 @@ export const SiteDefinition: ISiteDefinition<Platforms> = {
...
+    {
+      path: '#/controls/web',
+      exclude: 'fluent-theme',
+      text: 'You can now implement the new Fluent styles in Fabric Web controls.',
+      linkText: 'Learn more',
+      linkUrl: '#/controls/web/fluent-theme',
+      sessionStoragePrefix: 'WebFluentUpdates'
+    },
+    {
+      path: new RegExp(/^#?\/?$/),
```
Comment (jordandrako) [inline apps/fabric-website/src/SiteDefinition/SiteDefinition.tsx]: @ecraig12345 This is quite a bit cleaner. Still not ideal naming, but flexible.

File: apps/fabric-website/src/components/Site/Site.tsx [hunk 4]
```diff
@@ -219,27 +219,33 @@ export class Site<TPlatforms extends string = string> extends React.Component<IS
...
+        // Ensure path to match is a RegExp
+        path = new RegExp(path);
+        if (path.test(pagePath)) {
+          // The path matches, but test if there are exclusions that match
+          if (exclude !== undefined) {
+            // Ensure exclude is a RegExp
+            exclude = new RegExp(exclude);
+            if (exclude.test(pagePath)) {
+              // Exclude matched, break to return null
+              break;
```
Comment (jordandrako) [inline apps/fabric-website/src/components/Site/Site.tsx]: @ecraig12345 Can you think of a cleaner way to do exclusions than this?

**Pull request merged**